R-package/src/install.libs.R

```diff
@@ -163,12 +163,21 @@ if (!use_precompile) {
   }
 }
 
-# Check installation correctness
+# Packages with install.libs.R need to copy some artifacts into the
+# expected places in the package structure.
+# see https://cran.r-project.org/doc/manuals/r-devel/R-exts.html#Package-subdirectories,
+# especially the paragraph on install.libs.R
 dest <- file.path(R_PACKAGE_DIR, paste0("libs", R_ARCH), fsep = "/")
 dir.create(dest, recursive = TRUE, showWarnings = FALSE)
 if (file.exists(src)) {
   print(paste0("Found library file: ", src, " to move to ", dest))
   file.copy(src, dest, overwrite = TRUE)
+
+  symbols_file <- file.path(R_PACKAGE_SOURCE, "src", "symbols.rds")
```

Comment:
I think usage of `source_dir` here will simplify the code
https://github.com/microsoft/LightGBM/blob/6ba84d540c0132d4a603366fc91d4ccaa7b46d69/R-package/src/install.libs.R#L29

Reply:
Good point! I fixed a few cases of that here: https://github.com/microsoft/LightGBM/commit/1666f67859d2b0635c0b1fd45a26493319d8fbce

I also rebased to `master` to get recent changes